src/Components/Web/test/Forms/EditFormTest.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.AspNetCore.Components.RenderTree;
+using Microsoft.AspNetCore.Components.Test.Helpers;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    public class EditFormTest
+    {
+
+        [Fact]
+        public async Task ThrowsIfBothEditContextAndModelAreSupplied()
+        {
+            // Arrange
+            var editForm = new EditForm
+            {
+                EditContext = new EditContext(new TestModel()),
+                Model = new TestModel()
+            };
+            var testRenderer = new TestRenderer();
+            var componentId = testRenderer.AssignRootComponentId(editForm);
+
+            // Act/Assert
+            var ex = await Assert.ThrowsAsync<InvalidOperationException>(
+                () => testRenderer.RenderRootComponentAsync(componentId));
+            Assert.StartsWith($"{nameof(EditForm)} requires a {nameof(EditForm.Model)} parameter, or an {nameof(EditContext)} parameter, but not both.", ex.Message);
+        }
+
+        [Fact]
+        public async Task ReturnsEditContextWhenModelParameterUsed()
+        {
+            // Arrange
+            var model = new TestModel();
+            var rootComponent = new TestEditFormHostComponent
+            {
+                Model = model
+            };
+            var editFormComponent = await RenderAndGetTestEditFormComponentAsync(rootComponent);
+
+            // Act
+            var returnedEditContext = editFormComponent.EditContext;
+
+            // Assert
+            Assert.NotNull(returnedEditContext);
```

Comment:
```suggestion
            Assert.NotNull(returnedEditContext);
            Assert.Same(model, returnedEditContext.Model);
```

Reply:
Resolving this as it looks like @chrissainty has already made this update!